testsuite/tests/conftest.py

```diff
@@ -757,12 +759,17 @@ def finalizer():
 
 
 @pytest.fixture(scope="module")
-def backends_mapping():
+def backend_default(private_base_url, custom_backend):
+    return custom_backend("backend_default", endpoint=private_base_url())
```

Comment:
as far as I undertand this, this is irrelevant change to this PR. Even if necessary, better to have it in separate commit at least, better in separate PR (still difficult to accept, there are explicit tests for APIAP, however there are no explicit tests for legacy behavior, so this is something to discuss as it will reduce test coverage from certain point of view)